services/core/shared/core/compat/sdk_v5/blocks.js

```diff
@@ -90,17 +88,30 @@ const normalizeBlocks = async (blocks, includeGenesisAccounts = false) => {
 			block.totalBurnt = BigInt('0');
 			block.totalFee = BigInt('0');
 
-			block.payload.forEach(txn => {
-				txn.size = apiClient.transaction.encode(txn).length;
-				txn.minFee = apiClient.transaction.computeMinFee(txn);
-
-				block.size += txn.size;
-
-				const txnMinFee = BigInt(txn.minFee);
-				block.totalForged += BigInt(txn.fee);
-				block.totalBurnt += txnMinFee;
-				block.totalFee += BigInt(txn.fee) - txnMinFee;
-			});
+			await BluebirdPromise.map(
+				block.payload,
+				async (txn) => {
+					txn.size = apiClient.transaction.encode(txn).length;
+					txn.minFee = computeMinFee(
+						await getTxnAssetSchema(txn),
+						txn,
+						{
+							minFeePerByte: (await getGenesisConfig()).minFeePerByte,
+							baseFees: (await getGenesisConfig()).baseFees,
+							numberOfSignatures: txn.signatures.filter(s => s.length).length,
+							numberOfEmptySignatures: txn.signatures.filter(s => !s.length).length,
+						},
+					);
+
+					block.size += txn.size;
+
+					const txnMinFee = BigInt(txn.minFee);
+					block.totalForged += BigInt(txn.fee);
+					block.totalBurnt += txnMinFee;
+					block.totalFee += BigInt(txn.fee) - txnMinFee;
+				},
```

Comment:
Please wrap this logic into a named function.

Reply:
Since the current code logic also affects the `block` object while acting only on the block payload (transactions), refactored the code to calculate the transaction min fee into a named function instead.